dockerfiles/Dockerfile

```
@@ -0,0 +1,204 @@
+FROM nixos/nix:2.18.8
+
+# default build args
+ARG MAX_JOBS=4
+ARG CORES=4
+
+# Combine RUN commands to reduce layers and improve caching
+RUN echo "experimental-features = nix-command flakes" >> /etc/nix/nix.conf \
+    && echo "max-jobs = $MAX_JOBS" >> /etc/nix/nix.conf \
+    && echo "cores = $CORES" >> /etc/nix/nix.conf \
+    && nix profile install nixpkgs#cachix nixpkgs#git-lfs \
+    && cachix use kernel-builder
```

Just realized that these are not updated yet.

```suggestion
    && cachix use huggingface
```

Same change is needed in `Dockerfile.user`.